lib/topology/platforms/shell.py

```diff
@@ -359,23 +510,30 @@ def __init__(
             **kwargs
         )
 
-    def _setup_shell(self):
+    def _setup_connection(self, connection=None):
         """
         Overriden setup function that will disable the echo on the device on
         the shell and set a pexpect-safe prompt.
         """
+        if connection is None:
+            connection = self._default_connection
+
         # Wait initial prompt
-        self._spawn.expect(self._initial_prompt, timeout=self._timeout)
+        self._connections[connection].expect(
+            self._initial_prompt, timeout=self._timeout
+        )
 
         # Remove echo
-        self._spawn.sendline('stty -echo')
-        self._spawn.expect(self._initial_prompt, timeout=self._timeout)
+        self._connections[connection].sendline('stty -echo')
+        self._connections[connection].expect(
+            self._initial_prompt, timeout=self._timeout
+        )
 
         # Change prompt to a pexpect secure prompt
-        self._spawn.sendline(
+        self._connections[connection].sendline(
             'export PS1={}'.format(PExpectBashShell.FORCED_PROMPT)
         )
-        self._initial_prompt = self._prompt = PExpectBashShell.FORCED_PROMPT
```

Comment:
This is necessary, because a second (SSH, for example) connection will find the initial_prompt set to FORCED_PROMPT and fail its expect.